### PR TITLE
IP_Logic_UserRankInLocalStorage

### PR DESCRIPTION
**Pull request by skorpiom**
Hey,
Solution to this one:
 https://github.com/CodersCamp2020/CodersCamp2020.Project.JavaScript.StarWarsQuiz/issues/24

### CODE REVIEW

File: src/app/logic/UserRankInLocalStorage/UserRankInLocalStorage.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      || typeof obj.userName !== 'string'
+      || typeof obj.numberOfCorrectAnswers !== 'number'
+      || typeof obj.numberOfTotalAnswers !== 'number' ) throw Error('Incorrect type of arguments.');
+
+  //check if the mode of the game is correct
+  let isCorrectMode = false;
+  MODES.forEach(mode => {
+    if(obj.gameMode === mode) isCorrectMode = true;
+  });
+  if(!isCorrectMode) throw Error('Incorrect game mode.');
```
Comment (kjreczek) [inline src/app/logic/UserRankInLocalStorage/UserRankInLocalStorage.js:23]: [...].includes() would do the same but will be shorter and easier to read :P

File: src/app/logic/UserRankInLocalStorage/UserRankInLocalStorage.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      numberOfTotalAnswers: number
+    }
+    */
+
+function getUserRankInLocalStorage(gameMode) {
+  //check if the mode of the game is correct
+  let isCorrectMode = false;
+  MODES.forEach(mode => {
+    if(gameMode === mode) isCorrectMode = true;
+  });
```
Comment (kjreczek) [inline src/app/logic/UserRankInLocalStorage/UserRankInLocalStorage.js:65]: same here

File: test/UserRankInLocalStorage.spec.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+const obj = {
+  gameMode: 'People',
+  userName: 'test',
+  numberOfCorrectAnswers: 22,
+  numberOfTotalAnswers: 33,
+}
+
+let objIncorrect = Object.create(obj);
+    objIncorrect.gameMode = 2;
```
Comment (kjreczek) [inline test/UserRankInLocalStorage.spec.js:11]: did you mean to leave it here?

**Comment by skorpiom** [comment]
You're right, thanks! Changed.

**Review by juliakozlowska — APPROVED** [review]

**Review by kjreczek — APPROVED** [review]
👍

**Pull request merged**